cms/core/models/snippets.py

```diff
@@ -16,7 +15,7 @@
     from cms.users.models import User
 
 
-class ContactDetails(index.Indexed, models.Model):
+class ContactDetails(TranslatableMixin, index.Indexed, models.Model):
```

Comment:
🤔 we likely need https://docs.wagtail.org/en/stable/advanced_topics/i18n.html#making-snippets-with-existing-data-translatable here. Have you checked with existing `ContactDetail` instances?